lib/helper.js

```diff
@@ -0,0 +1,218 @@
+'use strict';
+
+const Hoek = require('@hapi/hoek');
+
+/**
+ * Convert job steps from array to object for faster lookup
+ * @method convertFromArrayToObject
+ * @param  {Array} steps Job step array of objects
+ * @return {Object}      Job object
+ */
+function convertFromArrayToObject(steps) {
+    return steps.reduce((obj, item) => {
+        const key = Object.keys(item)[0];
+
+        obj[key] = item[key];
+
+        return obj;
+    }, {});
+}
+/**
+ * Merge oldJob into newJob
+ * "oldJob" takes precedence over "newJob". For ex: child template job settings > parent template job settings
+ * @param  {Object}   newJob        Job to be merged into. For ex: parent template
+ * @param  {Object}   oldJob        Job to merge. For ex: child template
+ * @param  {Boolean}  fromTemplate  Whether this is merged from template. If true, perform extra actions such as wrapping.
+ */
+function merge(newJob, oldJob, fromTemplate) {
+    let warnings = [];
+
+    // Intialize new job with default fields (environment, settings, and secrets)
+    newJob.annotations = newJob.annotations || {};
+    newJob.environment = newJob.environment || {};
+    newJob.settings = newJob.settings || {};
+
+    // Merge
+    Object.assign(newJob.annotations, oldJob.annotations || {});
+    Object.assign(newJob.environment, oldJob.environment || {});
+    Object.assign(newJob.settings, oldJob.settings || {});
+    newJob.image = oldJob.image || newJob.image;
+
+    if (oldJob.requires) {
```

Comment:
are line 41 to line 55 necessary for merging templates?

Reply:
these are all fields that could exist in a template, right?

Reply:
cache might make sense, but for others I don't think anyone would expect them in templates.

Reply:
I think it's okay to leave it
if template owners choose to use it, template users just need to be aware